packages/asset-uploader/src/targets/s3/handler.ts

```
@@ -0,0 +1,67 @@
+import { PutObjectCommandInput, S3Client } from "@aws-sdk/client-s3";
+import { Upload } from "@aws-sdk/lib-storage";
+import ObjectID from "bson-objectid";
+
+import sharp from "sharp";
+import {
+  assetEnvVariables,
+  ImagesUploadedSuccess,
+  s3EnvVariables,
+  S3UploadHandler,
+} from "../../schema";
+import {
+  getArrayBufferFromIterable,
+  getFilenameAndExtension,
+} from "../../helpers/array-buffer-helpers";
+
+export const s3UploadHandler: S3UploadHandler = async ({
+  file: { data, filename: baseFileName, contentType },
+  maxPartSize,
+}) => {
+  const s3Envs = s3EnvVariables.parse(process.env);
+  const { MAX_UPLOAD_SIZE } = assetEnvVariables.parse(process.env);
+  if (!data) return;
```

when is data empty? is that normal or should we throw an error?